src/WebJobs.Script/Rpc/MessageExtensions/RpcMessageConversionExtensions.cs

```diff
@@ -189,9 +255,34 @@ public static TypedData ToRpc(this object value, ILogger logger, Capabilities ca
                     }
                 }
             }
-            else
+            return typedData;
+        }
+
+        public static TypedData ToRpc(this object value, ILogger logger, Capabilities capabilities)
+        {
```

Comment:
Tested the following, please take a look at the following: 
```
public static TypedData ToRpc(this object value, ILogger logger, Capabilities capabilities)
        {
            TypedData typedData = null;
            if (value == null)
            {
                typedData = new TypedData();
                return typedData;
            }
            if (value.GetType().IsArray && IsTypedDataCollectionSupported(capabilities))
            {
                typedData = value.ToRpcCollection();
            }
            if (typedData == null)
            {
                if (value is HttpRequest request)
                {
                    typedData = value.ToRpcHttp(logger, capabilities);
                }
                else
                {
                    typedData = value.ToRpcPrimitive();
                }
            }
            if (typedData == null)
            {
                // attempt POCO / array of pocos
                typedData = new TypedData();
                try
                {
                    typedData.Json = JsonConvert.SerializeObject(value);
                }
                catch
                {
                    typedData.String = value.ToString();
                }
            }
            return typedData;
        }
```

Reply:
@pragnagopa Thank you so much for the update, I would like to suggest to remove the if (value is HttpRequest request) for 1. we have this check in the ToRpcHttp. 2. Consistent with the paradigm for checking the typeData == null.

          public static TypedData ToRpc(this object value, ILogger logger, Capabilities capabilities)
        {
            TypedData typedData = null;
            if (value == null)
            {
                typedData = new TypedData();
                return typedData;
            }
            if (value.GetType().IsArray && IsTypedDataCollectionSupported(capabilities))
            {
                typedData = value.ToRpcCollection();
            }
            if (typedData == null)
            {
                    typedData = value.ToRpcHttp(logger, capabilities);
           }
           if (typedData == null)
            {
                    typedData = value.ToRpcPrimitive();
            }
            if (typedData == null)
            {
                // attempt POCO / array of pocos
                typedData = new TypedData();
                try
                {
                    typedData.Json = JsonConvert.SerializeObject(value);
                }
                catch
                {
                    typedData.String = value.ToString();
                }
            }
            return typedData;
        }

Reply:
There is different idea is instead of using the if and the same code again, what about to break each if condition to a function. And create a map the key is the type and value is the function. And let the map do the if condition for me. Sure this will need reflections :)

Reply:
Else is needed ... because if the value is array and types data collection is supported by the work ... no need to check other types..it should fall back to json

Reply:
We should not introduce reflection api without performance testing and is out of scope for this change

Reply:
I agree with you but currently we have byte[] as a primitive which we do not want to be jason, correct?

Reply:
Yes . Please add byte[] in to rpc  collection.